modules/samples/connectedcup/component/api/src/main/java/org/coffeeking/api/ConnectedCupServiceImpl.java

```diff
@@ -116,4 +123,38 @@ private String getSensorEventTableName(String sensorName) {
         return sensorEventTableName;
     }
 
+    @Path("device/register")
+    @POST
+    public boolean register(@QueryParam("name") String name) {
+        try {
+            DeviceIdentifier deviceIdentifier = new DeviceIdentifier();
+            String deviceId = shortUUID();
+            deviceIdentifier.setId(deviceId);
+            deviceIdentifier.setType(ConnectedCupConstants.DEVICE_TYPE);
+            if (APIUtil.getDeviceManagementService().isEnrolled(deviceIdentifier)) {
+                return false;
+            }
+            Device device = new Device();
+            device.setDeviceIdentifier(deviceId);
+            EnrolmentInfo enrolmentInfo = new EnrolmentInfo();
+            enrolmentInfo.setDateOfEnrolment(new Date().getTime());
+            enrolmentInfo.setDateOfLastUpdate(new Date().getTime());
+            enrolmentInfo.setStatus(EnrolmentInfo.Status.ACTIVE);
+            enrolmentInfo.setOwnership(EnrolmentInfo.OwnerShip.BYOD);
+            device.setName(name);
+            device.setType(ConnectedCupConstants.DEVICE_TYPE);
+            enrolmentInfo.setOwner(APIUtil.getAuthenticatedUser());
+            device.setEnrolmentInfo(enrolmentInfo);
+            return APIUtil.getDeviceManagementService().enrollDevice(device);
+        } catch (DeviceManagementException e) {
+            return false;
```

Comment:
Would have been helpful if the exception was logged before returning false for the ease of troubleshooting issues.